overlord/snapstate/snapmgr_test.go

```diff
@@ -930,28 +938,159 @@ func (s *snapmgrTestSuite) TestRemoveIntegration(c *C) {
 			name: "/snap/some-snap/7",
 		},
 		fakeOp{
+			op:    "remove-profiles:Doing",
+			name:  "some-snap",
+			revno: 7,
+		},
+		fakeOp{
 			op:   "remove-snap-data",
 			name: "/snap/some-snap/7",
 		},
 		fakeOp{
 			op:   "remove-snap-files",
 			name: "/snap/some-snap/7",
 		},
+		fakeOp{
+			op:   "discard-conns:Doing",
+			name: "some-snap",
+		},
 	}
 	c.Assert(s.fakeBackend.ops, DeepEquals, expected)
 
 	// verify snapSetup info
 	tasks := ts.Tasks()
-	// snap-setup is in discard-snap above discard-conns.
-	task := tasks[len(tasks)-2]
-	var ss snapstate.SnapSetup
-	err = task.Get("snap-setup", &ss)
-	c.Assert(err, IsNil)
-	c.Assert(ss, DeepEquals, snapstate.SnapSetup{
-		Name:     "some-snap",
-		Revision: 7,
+	for _, t := range tasks {
+		ss, err := snapstate.TaskSnapSetup(t)
+		c.Assert(err, IsNil)
+
+		var expSnapSetup *snapstate.SnapSetup
+		if t.Kind() == "discard-conns" {
+			expSnapSetup = &snapstate.SnapSetup{
+				Name: "some-snap",
+			}
+		} else {
+			expSnapSetup = &snapstate.SnapSetup{
+				Name:     "some-snap",
+				Revision: 7,
+			}
+		}
+		c.Check(ss, DeepEquals, expSnapSetup, Commentf(t.Kind()))
+	}
+
+	// verify snaps in the system state
+	var snapst snapstate.SnapState
+	err = snapstate.Get(s.state, "some-snap", &snapst)
+	c.Assert(err, Equals, state.ErrNoState)
+}
+
+func (s *snapmgrTestSuite) TestRemoveWithManyRevisionsIntegration(c *C) {
```

Comment:
:+1: this is good.
Do you think an additional user test would be useful here? One that calls snap install twice, snap remove once, and verifies that snap list doesn't show the snap.